hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -241,7 +241,12 @@ object DefaultSource {
     }
 
     if (metaClient.getCommitsTimeline.filterCompletedInstants.countInstants() == 0) {
-      new EmptyRelation(sqlContext, resolveSchema(metaClient, parameters, Some(schema)))
+      val structType = resolveSchema(metaClient, parameters, Some(schema))
```

Comment:
can you help me understand why we might get null schema while reading a hudi table ?

Reply:
Actually that part I'm not aware of. There is a database I have where this behaviour can be reproduced. I'll see if I can share that table after deanonymze or something.

Reply:
Perhaps this bit from my JIRA ticket can help? "If there are no completed instants in the table, and there is no user defined schema for it as well (as represented by the userSpecifiedSchema field in DataSource.scala), then the EmptyRelation returned by DefaultSource.createRelation sets schema of the EmptyRelation to null. This breaks the contract of Spark's BaseRelation, where the schema is a StructType but is not expected to be null."

Reply:
Since this seems to happen on an empty table with no completed commits, shall we keep this as a valid Hudi issue then? (The scope of my current task was only to validate if it's a valid issue). I can imagine situations like this where people will be querying a table before it is populated. I think it ought to return empty rows and not throw an NPE. What do you think?

Reply:
Issue has been noted as a valid bug in the JIRA: https://issues.apache.org/jira/browse/HUDI-5688?focusedCommentId=17688209&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17688209